src/test/ui/associated-types/higher-ranked-projection.bad.stderr

```diff
@@ -4,8 +4,8 @@ error[E0308]: mismatched types
 LL |     foo(());
    |     ^^^ lifetime mismatch
    |
-   = note: expected reference `&'a ()`
-                   found type `&()`
+   = note: expected type `&'a ()`
```

Comment:
Any idea why this one changed? Or for that,matter, why the one below always mentioned type instead of reference?

Reply:
🤷‍♂️ I have completely no idea, it should be exactly equivalent in most cases, maybe there's some non-exhaustive match buried deep somewhere?

Reply:
On a computer now, I think it is https://github.com/rust-lang/rust/blob/926e7843eaa1794f15948395588eddedfb74a0d8/compiler/rustc_infer/src/infer/error_reporting/mod.rs#L1585

this will get solved along with eliminating `ValuePair::Ty` in favour of `::Term`.